WalletWasabi.Tests/UnitTests/BlockNotifierTests.cs

```diff
@@ -77,16 +79,41 @@ public async Task NotifyBlocksAsync()
 
 			// Assert that the blocks come in the right order
 			var height = 0;
-			void OnBlockInv(object s, Block b) => Assert.Equal(b.GetHash(), chain.GetBlock(height++).HashBlock);
+			string message = string.Empty;
+
+			void OnBlockInv(object blockNotifier, Block b)
+			{
+				uint256 h1 = b.GetHash();
+				uint256 h2 = chain.GetBlock(height + 1).HashBlock;
+
+				if (h1 != h2)
+				{
+					message = string.Format("height={0}, [h1] {1} != [h2] {2}", height, h1, h2);
```

Comment:
@lontivero I'm not entirely happy with this. I would like to dump both blocks so that it would be clear what's the difference between the block if anything fails. Do you happen to know about any util class in NBitcoin that would help with that?

Reply:
I think dumping block is not a good idea. Even the json representation would be huge and hard to compare and in case we find an easy way to detect the difference between two blocks, what would that be good for?

Reply:
It took me quite a while to figure out why this test was failing. Having reasonable debug output would have made it easier. So dumping block may be an overkill but header is not that large..  Anyway, it was just a suggestion. It's not necessary to do it.